arduino-ide-extension/src/node/core-service-impl.ts

```diff
@@ -76,15 +76,19 @@ export class CoreServiceImpl extends CoreClientAware implements CoreService {
                 .filter(notEmpty)
                 .shift() ?? error.details
             );
-            this.sendResponse(
-              error.details + '\n\n' + message,
-              OutputMessage.Severity.Error
+            const chunk = new TextEncoder().encode(
+              `${error.details}'\n\n'${message}`
             );
+            handler.addChunk(chunk, OutputMessage.Severity.Error);
+
             reject(CoreError.VerifyFailed(message, compilerErrors));
           }
         })
-        .on('end', resolve);
-    }).finally(() => handler.dispose());
+        .on('end', () => {
+          handler.dispose();
```

Comment:
the `on('end'` callback will always be called even if there is an error. It fires after the `on('error'` callback, and signifies there are no more messages to be receive, hence doing `handler.dispose` on `.finally` is incorrect. As `.on('error'` will reject the promise, we may get to the `.finally` callback before the `on('end'` callback is called, in this case messages recieved after the error would not be captured